openhcl/openhcl_boot/src/arch/x86_64/tdx.rs

```diff
@@ -3,16 +3,90 @@
 
 //! TDX support.
 
+use crate::arch::x86_64::address_space::tdx_share_large_page;
+use crate::arch::x86_64::address_space::tdx_unshare_large_page;
+use crate::host_params::PartitionInfo;
+use crate::hvcall;
 use crate::single_threaded::SingleThreaded;
 use core::arch::asm;
 use core::cell::Cell;
+use loader_defs::shim::TdxTrampolineContext;
 use memory_range::MemoryRange;
 use safe_intrinsics::cpuid;
 use tdcall::AcceptPagesError;
 use tdcall::Tdcall;
 use tdcall::TdcallInput;
 use tdcall::TdcallOutput;
+use tdcall::tdcall_hypercall;
 use tdcall::tdcall_map_gpa;
+use tdcall::tdcall_wrmsr;
+use x86defs::tdx::RESET_VECTOR_PAGE;
+use x86defs::tdx::TdVmCallR10Result;
+
+/// Writes a synthehtic register to tell the hypervisor the OS ID for the boot shim.
+pub fn report_os_id(guest_os_id: u64) {
+    tdcall_wrmsr(
+        &mut TdcallInstruction,
+        hvdef::HV_X64_MSR_GUEST_OS_ID,
+        guest_os_id,
+    )
+    .unwrap();
+}
+
+/// Initialize hypercalls for a TDX L1, sharing the hypercall I/O pages with the HV
+///
+/// # Safety
+///
+/// The caller ensures that the I/O pages are valid and not concurrently accessed
+pub unsafe fn initialize_hypercalls(guest_os_id: u64, input_page: u64, output_page: u64) {
+    // TODO We are assuming we are running under a Microsoft hypervisor, so there is
+    // no need to check any cpuid leaves.
+    report_os_id(guest_os_id);
+
+    // SAFETY: The hypercall i/o pages are valid virtual addresses owned by the caller
```

Comment:
i would again elaborate a bit more on these safety calls, based on what share page is doing.